llvm/lib/Target/AArch64/GISel/AArch64CallLowering.cpp

```diff
@@ -1293,8 +1293,19 @@ bool AArch64CallLowering::lowerCall(MachineIRBuilder &MIRBuilder,
            !Subtarget.noBTIAtReturnTwice() &&
            MF.getInfo<AArch64FunctionInfo>()->branchTargetEnforcement())
     Opc = AArch64::BLR_BTI;
-  else
+  else {
+    // For an intrinsic call (e.g. memset), use GOT if "RtLibUseGOT" (-fno-plt)
+    // is set.
+    if (Info.Callee.isSymbol() && F.getParent()->getRtLibUseGOT()) {
+      auto Reg =
+          MRI.createGenericVirtualRegister(getLLTForType(*F.getType(), DL));
+      auto MIB = MIRBuilder.buildInstr(TargetOpcode::G_GLOBAL_VALUE);
+      DstOp(Reg).addDefToMIB(MRI, MIB);
```

Comment:
use buildGlobalValue

Reply:
This part is to handle library calls.
`buildGlobalValue` builds a `G_GLOBAL_VALUE` with `MIB.addGlobalAddress(GV);`. Here we use an ExternalSymbol, and cannot use `buildGlobalValue`

Reply:
Missing overload then, should try to avoid raw buildInstr calls when possible

Reply:
Did you propose a new MachineIRBuilder API? This is a special use of `buildInstr(TargetOpcode::G_GLOBAL_VALUE)` and there is only one. I think avoid adding another API is good for now.

Reply:
`buildInstr(TargetOpcode::G_GLOBAL_VALUE)` cannot be CSEed today, and I think this patch should not change it.

It seems that only the `buildInstr` overload with `DstOps/SrcOps` can perform CSE. `MachineIRBuilder::buildGlobalValue` does not call this overload.

Reply:
Using raw buildInstr is a worse API. Please just add the new overload

Reply:
I disagree. I think adding a function similar to `buildGlobalValue`, either omits `const GlobalValue *GV` or changes the argument to `const char*`, will add confusion, as this is a very special use case.

I slightly simplified the code in the just-pushed commit.